panic on asset scan (fs scan)

**Describe the bug**

```
→ run policies for asset asset=i-032302d0105f9c9c8
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x4a90f20]
goroutine 127 [running]:
go.mondoo.com/cnquery/resources/packs/os/smbios.(*LinuxSmbiosManager).Info.func1({0x6eb3ab2, 0x12}, {0x0, 0x0}, {0x150?, 0x69104c0?})
/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20221012222849-08b70aa0032c/resources/packs/os/smbios/linux.go:29 +0x60
```

**To Reproduce**
this was a filesystem scan (`mondoo scan aws ec2 ebs instanceid`) on an ec2 instance in aws (`ubuntu/images/hvm-ssd/ubuntu-jammy-22.04-amd64-server-20220609`)

--- a/resources/packs/os/smbios/linux.go
+++ b/resources/packs/os/smbios/linux.go
@@ -1,7 +1,7 @@
 package smbios
 
 import (
-	"io/ioutil"
+	"io"
 	"os"
 	"path/filepath"
 	"strings"
@@ -26,6 +26,11 @@ func (s *LinuxSmbiosManager) Info() (*SmBiosInfo, error) {
 	root := "/sys/class/dmi/id/"
 
 	wErr := afs.Walk(root, func(path string, info os.FileInfo, fErr error) error {
+		if fErr != nil {
+			// we skip files that we cannot access
+			return filepath.SkipDir
+		}
+
 		if info.IsDir() && path != root {
 			return filepath.SkipDir
 		}
@@ -80,7 +85,7 @@ func (s *LinuxSmbiosManager) Info() (*SmBiosInfo, error) {
 				return err
 			}
 			defer f.Close()
-			data, err := ioutil.ReadAll(f)
+			data, err := io.ReadAll(f)
 			if err != nil {
 				return err
 			}

